### PR TITLE
FRONT-33

### PR DESCRIPTION
**Pull request by Ekaterina-Bondareva**
[Create Address Adapter#33](https://github.com/Code-the-Dream-School/bb-practicum-team1-front/issues/33)

### CODE REVIEW

File: src/adapters/address-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    fetch(`https://{process.env.HOSTNAME}/api/v1/autocomplete/search?address=${address}`, {
+        method: 'get',
+    })
+    .then((resp) => {
+        let result: Array<Address> = <Array<Address>><unknown>resp.json();
+        return result;
+    })
+    .catch((e) => {
+        console.log(e);
+    });
```
Comment (koral14) [inline src/adapters/address-adapter.ts:45]: Nicely done Ekaterina! I'm not sure I can advise you better.

**Review by cecerodrgz — APPROVED** [review]

File: src/adapters/address-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * getAddressCoordinate(address);
+ * @returns {Promise<void | Address>} A promise that is top matching result for given address with coordinates..
+ */
+export const getAddressCoordinate = (address: string): Promise<void | Address> => 
+    // Call backend API to search for given address. ${process.env.HOSTNAME} variable is used to avoid hardcoding hostname
+    // to allow running it locally or remotly without code changes.
+    // Value for HOSTNAME is provided in .env file
+    fetch(`https://{process.env.HOSTNAME}/api/v1/address/search?address=${address}`, {
+        method: 'get',
+    })
```
Comment (jpremo) [inline src/adapters/address-adapter.ts:22]: For this fetch, and the one below, import and use fetchAPIData like the other adapter files do. You will also be able to get rid of resp.json.

File: src/adapters/address-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * This function will return the best match for given address with coordinates.
+ * @param {String} address - address to search.
+ * @example
+ * const address = "315 Kildaire Farm Rd, Cary, NC 27511";
+ * getAddressCoordinate(address);
+ * @returns {Promise<void | Address>} A promise that is top matching result for given address with coordinates..
+ */
+export const getAddressCoordinate = (address: string): Promise<void | Address> =>
```
Comment (jpremo) [inline src/adapters/address-adapter.ts:16]: Can you convert both adapters to async/await for consistency with the other adapters?

**Review by jpremo — CHANGES_REQUESTED** [review]

File: src/adapters/address-adapter.ts (hunk 2 above)
Comment (Ekaterina-Bondareva) [reply src/adapters/address-adapter.ts]: done

File: src/adapters/address-adapter.ts (hunk 3 above)
Comment (Ekaterina-Bondareva) [reply src/adapters/address-adapter.ts]: done

**Review by jpremo — APPROVED** [review]

**Pull request merged**